mysite/ct/views.py

```diff
@@ -155,8 +155,8 @@ def fsm_redirect(self, request, eventName=None, defaultURL=None,
         # now we check here whether event is actually from path matching
         # this node
         referer = request.META.get('HTTP_REFERER', '')
-        referer = referer[referer.find('/ct/'):]
-        if request.method == 'POST' and referer != self.fsmStack.state.path \
+        origin = request.META.get('HTTP_ORIGIN', '')
```

Comment:
I now see that this bug is the cause of all the strange behaviors I've seen today, because it is screwing up handling of all 'next' events if the browser does not send the HTTP_ORIGIN header (and my firefox 38 does not).